File: backend/apps/schools/urls.py
from django.urls import path

from .views import (
    DepartmentDetailView,
    DepartmentListCreateView,
    SchoolBySlugView,
    SchoolDetailView,
    SchoolListCreateView,
)

urlpatterns = [
    path("schools/", SchoolListCreateView.as_view(), name="school-list-create"),
    path("schools/by-slug/<slug:slug>/", SchoolBySlugView.as_view(), name="school-by-slug"),
    path("schools/<looseid:school_id>/", SchoolDetailView.as_view(), name="school-detail"),
    path("schools/<looseid:school_id>/departments/", DepartmentListCreateView.as_view(), name="department-list-create"),
    path("departments/<looseid:department_id>/", DepartmentDetailView.as_view(), name="department-detail"),
]

File: academia-frontend/lib/urls.js
import { slugify } from "./slugify";

export function questionUrl(question) {
  if (!question?.id) return "/questions";
  const slug = question.slug || (question.title ? slugify(question.title) : "");
  return slug ? `/questions/${question.id}/${slug}` : `/questions/${question.id}`;
}

export function schoolUrl(school) {
  if (!school?.id) return "/schools";
  const slug = school.slug || (school.name ? slugify(school.name) : "");
  return slug ? `/schools/${school.id}/${slug}` : `/schools/${school.id}`;
}